packages/code-snippet/src/CodeSnippetWidget.tsx

```diff
@@ -247,10 +251,52 @@ class CodeSnippetDisplay extends MetadataDisplay<ICodeSnippetDisplayProps> {
     );
   }
 
+  static getDerivedStateFromProps(
+    props: IMetadataDisplayProps,
+    state: IMetadataDisplayState
+  ): IMetadataDisplayState {
+    if (state.searchValue === '' && state.filterTags.length === 0) {
+      return {
+        metadata: props.metadata,
+        searchValue: '',
+        filterTags: []
+      };
+    }
+
+    if (state.searchValue !== '' || state.filterTags.length !== 0) {
+      const newMetadata = props.metadata.filter((metadata: IMetadata) => {
+        return (
+          (state.searchValue !== '' &&
+            metadata.name.toLowerCase().includes(state.searchValue)) ||
+          metadata.display_name.toLowerCase().includes(state.searchValue) ||
+          metadata.metadata.language
+            .toLowerCase()
+            .includes(state.searchValue) ||
+          (metadata.metadata.tags &&
+            metadata.metadata.tags.some((tag: string) =>
+              state.filterTags.includes(tag)
+            ))
+        );
+      });
```

Comment:
These checks won't cover all the edge cases, plus whenever I tried filtering it by a tag it didn't work (eg. I had only one code snippet tagged but all of them were still being displayed). 
![image](https://user-images.githubusercontent.com/25207344/97236354-493d2800-17bb-11eb-8b8b-48ebafca3a5a.png)

A suggested fix (tested and it works) and some refactoring could be:
```suggestion
      const filterTags = new Set(state.filterTags);
      const newMetadata = props.metadata.filter((metadata: IMetadata) => {
        const searchValue = state.searchValue.toLowerCase().trim();
        return searchValue
            && (metadata.name.toLowerCase().includes(searchValue) 
                || metadata.display_name.toLowerCase().includes(searchValue) 
                || metadata.metadata.language.toLowerCase().includes(searchValue)) 
            || metadata.metadata.tags 
                && metadata.metadata.tags.some((tag: string) => filterTags.has(tag));
      });
```